### PR TITLE
Move parsers and serializers to common

### PR DESCRIPTION
**Pull request by ewoutkramer**
By removing the dependency on ModelInfo, I have been able to move classes for parsing and serializing (at least a major part) to the common assemblies.  This is the low-hanging fruit, I'll see if there's more to move later to not make this PR overly big and complex.

### CODE REVIEW

File: src/Hl7.Fhir.Support/Utility/Cache.cs [hunk 1]
```diff
@@ -60,38 +62,59 @@ private void CopyTo(CacheSettings other)
...
             }
+            else
+            {
+                var cachedItem = _cached.GetOrAdd(key, _retriever);
+                enforceMaxItems();
+                return cachedItem.Value;
+            }
+        }
+
+        public V GetValueOrAdd(K key, V value)
```
Comment (marcovisserFurore) [inline src/Hl7.Fhir.Support/Utility/Cache.cs:109]: Because these are public functions, we should also document this

**Review by marcovisserFurore — CHANGES_REQUESTED** [review]
See my review below

File: src/Hl7.Fhir.Support.Poco/ElementModel/PocoBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+using System.Reflection;
+
+namespace Hl7.Fhir.Serialization
+{
+    internal class PocoBuilder : IExceptionSource
+    {
+        public PocoBuilder(ModelInspector inspector, PocoBuilderSettings settings = null)
+        {
+            _settings = settings?.Clone() ?? new PocoBuilderSettings();
+            _inspector = inspector;
```
[the remark this answers is not in the document]
Comment (ewoutkramer) [reply src/Hl7.Fhir.Support.Poco/ElementModel/PocoBuilder.cs:25]: we should!

File: src/Hl7.Fhir.Support.Poco/Introspection/PropertyMapping.cs [hunk 3]
```diff
@@ -90,15 +90,15 @@ public Type ElementType
...
+        /// attributes (e.g. Extension.url).
         /// </para>
         /// </remark>
         public Type[] FhirType { get; private set; }        // may be multiple if this is a choice
 
         /// <summary>
         /// True when the element is of type '*', e.g. Extension.value[x]. Any type is allowed.
         /// </summary>
-        public bool IsOpen { get; private set; }
+        //public bool IsOpen { get; private set; }
```
[the remark this answers is not in the document]
Comment (ewoutkramer) [reply src/Hl7.Fhir.Support.Poco/Introspection/PropertyMapping.cs:101]: That's right - I added it to the "breaking changes" document.  It had been introduced in 1.8 - but it was a bad idea.

File: src/Hl7.Fhir.Support.Tests/Introspection/PropertyMappingTest.cs [hunk 4]
```diff
@@ -68,6 +68,33 @@ public void TestVersionSpecificMapping()
             Assert.AreEqual(typeof(Canonical), profile.FhirType.Single());
         }
 
+        [TestMethod]
+        public void TestPropsWithRedirect()
+        {
+            _ = ClassMapping.TryCreate(typeof(TypeWithCodeOfT), out var mapping);
```
[the remark this answers is not in the document]
Comment (ewoutkramer) [reply src/Hl7.Fhir.Support.Tests/Introspection/PropertyMappingTest.cs:74]: ok.

File: src/Hl7.Fhir.Support.Tests/Utility/CacheTests.cs [hunk 5]
```diff
@@ -17,41 +19,98 @@ public class CacheTests
...
+            var cache = new Cache<string, string>(factory, new CacheSettings() { MaxCacheSize = 10 });
+
+            var value = cache.GetValue($"item");
+            Assert.AreEqual($"ITEM", value);
+            Assert.AreEqual(1, called);
+
+            called = 0;
+
+            var secondValue = cache.GetValue($"item");
+            Assert.IsTrue(object.ReferenceEquals(value,secondValue));
```
[the remark this answers is not in the document]
Comment (ewoutkramer) [reply src/Hl7.Fhir.Support.Tests/Utility/CacheTests.cs:39]: I am never sure with these Equals/Same/Equivalent calls. Find myself having to read the documentation over and over again, so I prefer this, it's more explicit. I *really* want reference equality....

File: src/Hl7.Fhir.Support/Utility/Cache.cs (hunk 1 above)
Comment (ewoutkramer) [reply src/Hl7.Fhir.Support/Utility/Cache.cs:109]: yes!

File: src/Hl7.Fhir.Support.Tests/Introspection/ClassMappingTest.cs [hunk 6]
```diff
@@ -10,12 +10,56 @@
 using Microsoft.VisualStudio.TestTools.UnitTesting;
 using Hl7.Fhir.Model;
 using Hl7.Fhir.Introspection;
+using System.Linq;
 
 namespace Hl7.Fhir.Tests.Introspection
 {
     [TestClass]
     public class ClassMappingTest
```
[the remark this answers is not in the document]
Comment (ewoutkramer) [reply src/Hl7.Fhir.Support.Tests/Introspection/ClassMappingTest.cs:18]: done!

File: src/Hl7.Fhir.Support.Poco/Validation/AllowedTypesAttribute.cs [hunk 7]
```diff
@@ -28,8 +28,6 @@ public AllowedTypesAttribute(params Type[] types)
        
         public Type[] Types { get; set; }
 
-        public bool IsOpen { get; set; }
```
[the remark this answers is not in the document]
Comment (ewoutkramer) [reply src/Hl7.Fhir.Support.Poco/Validation/AllowedTypesAttribute.cs:31]: Yes. Added this to the breaking changes wiki page.

**Review by marcovisserFurore — APPROVED** [review]

**Pull request merged**